### DISCUSSION TITLE
VE.Direct an OpenDTU-OnBattery (mit OpenDTU Fusion 2.1 Board)

### DISCUSSION THREAD

**Discussion by eMaitcH75** [hoylabs/OpenDTU-OnBattery#1090] [Q&A]
Guten Abend!
Je mehr ich lese, desto verwirrter werden ich. Deshalb muss ich hier mal fragen:
Aktuell hab ich nen "Victron SmartSolar MPPT VE.Can 150/70 rev2" über nen ESP32 mit VE.Direct (https://github.com/KinDR007/VictronMPPT-ESPHOME) an Home Assistant angebunden.
Jetzt habe ich OpenDTU Fusion - Board mit OpenDTU-OnBattery an einen Hoymils HMS-1600 4T angeschlossen und würde gerne die Funktion des ESP32 mit dem Fusion - Board ablösen. Hab heute schon gebastelt, aber ich bekomme einfach keine Werte im OpenDTU-OnBattery über die VE.Direct - Schnittstelle angezeigt.
Habe schon alles mögliche versucht. :-(

Der Anschluss vom MPPT geht an einen Logic Level Converter (siehe Anhang) und dann an den ESP (oder an das Fusion - Board). Habe es auch schon ohne diesen versucht..
Jetzt habe ich irgendwo gelesen, dass das Fusion - Board diesen Anschluss gar nicht unterstützt und man dafür dieses CAN/Iso Shield benötigt (https://github.com/markusdd/OpenDTUFusionDocs/blob/main/CANIso.md). 
Ist dem so?
In den Profilen der "Hardware-Einstellungen" auf der GUI von OpenDTU-OnBattery steht beim Eintrag "Victron" folgendes:
rx		-1
rx2		-1
tx		-1
tx2		-1
Das deutet ja auch drauf hin, dass der Anschluss nicht konfiguriert / geschaltet ist?!?

Eingesetzter Logic Converter:
![grafik](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/74836851/aeb7e647-e197-4d46-89c9-01ee708ccb32)

Wäre für ne Info sehr dankbar....

Michael

**Comment by AndreasBoehm** [comment]
> Jetzt habe ich irgendwo gelesen, dass das Fusion - Board diesen Anschluss gar nicht unterstützt und man dafür dieses CAN/Iso Shield benötigt (https://github.com/markusdd/OpenDTUFusionDocs/blob/main/CANIso.md). Ist dem so? 

Das ist korrekt, du musst entweder das `CAN/Iso Shield` besorgen und benutzen oder deinen `Logic Level Converter` korrekt anschließen und in OpenDTU konfigurieren.


> In den Profilen der "Hardware-Einstellungen" auf der GUI von OpenDTU-OnBattery steht beim Eintrag "Victron" folgendes: rx -1 rx2 -1 tx -1 tx2 -1 Das deutet ja auch drauf hin, dass der Anschluss nicht konfiguriert / geschaltet ist?!?

PIN `-1` bedeutet das dieser PIN nicht konfiguriert ist.

**Reply by eMaitcH75** [reply]
Danke für die schnelle Antwort! 
Ich gehe mal davon aus, dass ich den Logic Level Converter korrekt angeschlossen habe. 
Nehmen wir mal an, das wäre so. Dann würde ich in der "pin_mapping.json" (https://github.com/helgeerbe/OpenDTU-OnBattery/wiki/Device-Profiles) den Port 44 (aus https://github.com/markusdd/OpenDTUFusionDocs) wie folgt konfigurieren (aus https://github.com/helgeerbe/OpenDTU-OnBattery/wiki/VE.Direct-Interface)?

  "victron": {
       "rx": 44,
       "tx": -1,
       "rx2": 43,
       "tx2": -1,
   }

Wäre "victron" der korrekte Wert?

**Reply by schlimmchen** [reply]
Was du da zeigst ist plausibel. Ist natürlich nur ein Ausschnitt, oder? Was ist denn dein Stand? Hast du überhaupt zwei Victron Laderegler? Und hast du `generic_esp32_usb` (das `_usb` am Ende ist wichtig) installiert? Sonst sind pins 43/43 belegt durch die serielle Konsole.

**Reply by eMaitcH75** [reply]
> Was du da zeigst ist plausibel. Ist natürlich nur ein Ausschnitt, oder? Was ist denn dein Stand? Hast du überhaupt zwei Victron Laderegler? Und hast du `generic_esp32_usb` (das `_usb` am Ende ist wichtig) installiert? Sonst sind pins 43/43 belegt durch die serielle Konsole.

Ja, ist nur ein Ausschnitt aus dem verlinkten Beitrag. Ich nutze aktuell die "pin_mapping.json" - default Datei und der Konfiguration "OpenDTU Fusion v2 with CMT2300A and NRF24". Also folgendes:

`{
        "name": "OpenDTU Fusion v2 with CMT2300A and NRF24",
        "nrf24": {
            "miso": 48,
            "mosi": 35,
            "clk": 36,
            "irq": 47,
            "en": 38,
            "cs": 37
        },
        "cmt": {
            "clk": 6,
            "cs": 4,
            "fcs": 21,
            "sdio": 5,
            "gpio2": 3,
            "gpio3": 8
        },
        "eth": {
            "enabled": false,
            "phy_addr": -1,
            "power": -1,
            "mdc": -1,
            "mdio": -1,
            "type": 0,
            "clk_mode": 0
        },
        "led": {
            "led0": 17,
            "led1": 18
        }
    },`

Laderegler nutze ich aktuell nur einen, wieso zwei? 

Hmm Das "generic_esp32_usb" könnte natürlich der Fehler sein. Phuu. Ich mache mal ein Update auf die https://github.com/helgeerbe/OpenDTU-OnBattery/releases/tag/2024.06.03/opendtu-onbattery-generic_esp32s3_usb.bin und teste das nochmal. Das wäre natürlich doof von mir. :-/

Wenn ich die  opendtu-onbattery-generic_esp32s3_usb.bin installiert habe, muss ich die config aber dennoch wie oben erwähnt (pin_mapping.json) erweitern, oder?

Danke!

**Reply by schlimmchen** [reply]
Wenn du nur einen Laderegler hast, dann sind rx2 und tx2 natürlich -1. Und wie erwähnt sind pins 43/44 belegt, wenn du nicht die `_usb` Variante der Firmware installierst.

**Reply by akihilananda** [reply]
Mit opendtu Fusion und CAN/Iso Shield 
pin_mapping.json
battery rx 11, tx -1 (victron smart shunt)
victron rx 10, tx -1 (victron mppt 100 20)

**Comment by haegar33** [comment]
Nachdem das FusionBoard jetzt auch BLE 5 an Board hat wäre es theoretisch doch auch möglich die Verbindung zu den Victron Ladern auch über Bluetooth zu realisieren? Zu viel Aufwand?

**Reply by schlimmchen** [reply]
Bisher ausgeschlossen, weil der Code für BLE die bisherigen Grenzen der Flashpartition sofort gesprengt hat. Neuerdings, nachdem dieser Platz verdoppelt wurde (bzw. im nächsten Release werden wird) ist das denkbar. Wie groß der Aufwand ist, ist schwer abschätzbar. Ich zumindest hab kein Interesse dran.

Hast du schon gesehen, dass die mehrere Laderegler in ein VE.Smart Netzwerk packen kannst und dass du mit VE.Direct (TX benötigt) auch die Netzwerk-Gesamtleistung haben kannst, wenn du nur einen Laderegler mit OpenDTU-OnBattery verbindest?

**Reply by haegar33** [reply]
Ok, dachte ich mir schon das der Aufwand nicht unerheblich ist.
Was das "V.E. Smart Netzwerk" angeht so überlege ich mir für meine Situation ein Setup was langsam wachsen soll: bisher habe ich nur ein(!) 430Wp Panel, einen Hoymiles 600 und einen "alten"  LPF 12V Chinaakku. Über mehrere Victron Lader machen ich mir also vorerst keine Gedanken ;)

**Comment by eMaitcH75** [comment]
So, heute nochmal Zeit zum basteln gehabt..
Nachdem ich die Firmware **generic_esp32_usb** auf das Fusion- board geflashed, den Logic Level Converter mit den Kabeln (und dem Wiederstand) verlötet, die pin_mapping.json angepasst und "installiert", VE.Direct in OpenDTU-OnBattery aktiviert habe, läuft es.
Verrückt. :-)

**Comment by akihilananda** [comment]
generic_esp32_usb läuft auf dem Fusion board? 

Das Fusion board hat einen  ESP32-S3-WROOM-1U Mikrocontroller

Ich verwende deshalb opendtu-generic_esp32s3.factory.bin 
Mit OpenDTU Fusion CAN/Iso Shield wird Victron MPPT und smart shunt erkannt :-)

**Reply by schlimmchen** [reply]
> generic_esp32_usb läuft auf dem Fusion board?

Nein. Wahrscheinlich liegt der Tippfehler am fehlenden `s3` und nicht am `_usb` zuviel, d.h. gemeint war vermutlich `generic_esp32s3_usb`.

**Reply by eMaitcH75** [reply]
Ja, Tippfehler.. Hatte ich oben rauskopiert. Ist natürlich "opendtu-generic_esp32s3_usb.bin" aus https://github.com/tbnobody/OpenDTU/releases
Smart Shunt habe ich nicht, aber der "Smart Battery Sense" (Temperatur) wird auch erkannt...
MQTT Werte kommen im Home Assistant an, muss ich jetzt nur noch "richtig" implementieren.

Danke an alle!

**Reply by akihilananda** [reply]
OK. Ich bin jetzt etwas verwirrt.
Bei mir funktioniert das generic_esp32s3_usb_factory. bin nicht auf dem Fusion board V2.1

Wieso wird dann unterschieden zwischen:
opendtu-onbattery-generic_esp32s3_usb.factory.bin 
und
opendtu-onbattery-generic_esp32s3_factory.bin 
Sind beide für das Fusion Bord? Wo ist der Unterschied?
Danke.

**Reply by eMaitcH75** [reply]
Auf https://github.com/markusdd/OpenDTUFusionDocs/blob/main/BLANK_START.md steht folgendes:
"For the Fusion board, you need the USB version (as we use the in-built USB and no external converter)"
Dort wird auch das mit dem .factory - erklärt..

Was das allerdings mit dem seriellen Anschluss (VE.Direct) zu tun hat, kann ich auch nicht sagen.

**Reply by akihilananda** [reply]
Ja stimmt.
Vielen Dank für den Hinweis USB version funktioniert.:-)
